### PR TITLE
feat(dapps): Add connector badge to WC and BC dapps

### LINKED ISSUE #16815: [Browser plugin connector] Add the connector badge to dapp icon
#### [Figma](https://www.figma.com/design/1OYKMzU6KTQHQAqDhojk0r/Status-connector?node-id=2104-4428&node-type=frame&t=81HBj00gILQzjbx6-0)

Image

Image

Acceptance criteria:

- [ ] Each dapp in the dapps list has the connector badge
- [ ] On the connect modal, the dapp icon has the connector badge

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do

Closes #16815

- Adding connector badge to the dapps list and connect modal
- Updating the RoundImageWithBadge to support custom badge size and margins
- Updating the RoundImageWithBadge to support both SVG and PNG as source
- Polish the dapps sign modal badge to match the design
- Polish the dapps delegate in the dapps combo box to match the design
- Amend Storybook and qml tests

### Affected areas

Wallet connect
Browser connect

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/user-attachments/assets/c0c307b0-30a9-4458-bcdc-f8465f03671f

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsModule.qml [hunk 1]
```diff
@@ -220,7 +220,7 @@ SQUtils.QObject {
 
         onConnectDApp: (chains, dAppUrl, dAppName, dAppIcon, key) => {
             siwePlugin.connectionRequests.set(key.toString(), {chains, dAppUrl, dAppName, dAppIcon})
-            root.connectDApp(chains, dAppUrl, dAppName, dAppIcon, key)
+            root.connectDApp(chains, dAppUrl, dAppName, dAppIcon, Constants.WalletConnect, key)
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/services/dapps/DAppsModule.qml:223]:
```suggestion
            root.connectDApp(chains, dAppUrl, dAppName, dAppIcon, Constants.DAppConnectors.WalletConnect, key)
```

File: ui/app/AppLayouts/Wallet/services/dapps/plugins/DAppConnectionsPlugin.qml [hunk 2]
```diff
@@ -140,7 +140,7 @@ SQUtils.QObject {
         function onSessionProposal(sessionProposal) {
             const key = sessionProposal.id
             d.activeProposals.set(key.toString(), { context: sessionProposal, promise: bcConnectionPromise })
-            root.newConnectionProposed(key, [1], sessionProposal.params.proposer.metadata.url, sessionProposal.params.proposer.metadata.name, sessionProposal.params.proposer.metadata.icons[0])
+            root.newConnectionProposed(key, [1], sessionProposal.params.proposer.metadata.url, sessionProposal.params.proposer.metadata.name, sessionProposal.params.proposer.metadata.icons[0], Constants.StatusConnect)
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/services/dapps/plugins/DAppConnectionsPlugin.qml:143]:
```suggestion
            root.newConnectionProposed(key, [1], sessionProposal.params.proposer.metadata.url, sessionProposal.params.proposer.metadata.name, sessionProposal.params.proposer.metadata.icons[0], Constants.DAppConnectors.StatusConnect)
```

**Comment by alexjba** [comment]
> @alexjba what's the first ui in the video you used to test this please?

https://metamask.github.io/test-dapp/

**Review by caybro — APPROVED** [review]
LGTM

Just some minor stuff

**Review by virginiabalducci — APPROVED** [review]

**Pull request merged**